### ISSUE TITLE
Sync source-only branch on check-in

### ISSUE THREAD

**Issue by mmitche** [dotnet/source-build#3997]
When updates are made to a primary VMR branch, the source-only branch should get an automatic merge, with cloaking applied.

**Comment by premun** [comment]
As agreed, I have prepared the new VMR branch `main-ub` - https://github.com/dotnet/dotnet/tree/main-ub.
I have deleted all contents of the VMR and initialized it using the tooling without the cloakings. I then squashed these changes.
The result is this commit (on top of main): https://github.com/dotnet/dotnet/commit/ef0889835b8296b0bc3614c826561e17d257370a which adds 663 binaries (patch size for this would be 66.7MB).

The synchronization pipeline now runs 2 jobs:
https://dev.azure.com/dnceng/internal/_build/results?buildId=2363056&view=results

There is no CI running on this branch atm, I expect the VB guys to add their own pipelines and trigger on this branch.
